lib/approval.rb

```diff
@@ -18,23 +18,23 @@ def self.call
     yield approval_api
   rescue ApprovalApiClient::ApiError => e
     Rails.logger.error("ApprovalApiClient::ApiError #{e.message}")
-    raise Catalog::ApprovalError, e.message
+    raise Catalog::ApprovalError.new(e.code, e.message)
```

Comment:
@lindgrenj6 Shouldn't this also be an e.add

Reply:
No, it has to be a `Catalog::ApprovalError.new` because the comma syntax doesn't work with multiple arguments. We haven't created the `ManageIQ::API::Common::ApiError` object yet at this point, all we have is the `ApprovalApiClient::ApiError`. 

We do the initial `.new` on our specific exception and then from there one can add errors as necessary.

Reply:
@lindgrenj6 Shouldn't this be an attribute in the request object where we would keep an array of exceptions and at the end of the request check if its an empty list and if it has one or more error object we would send that whole list of error objects

Reply:
I'm not sure I get what you're saying here - we're rescuing if the generated OpenAPI client throws an exception. We don't really have control over what fields are on that request object, thats why I'm raising a new exception, the initialize method takes the same args and calls `.add()`: https://github.com/ManageIQ/manageiq-api-common/blob/83aa89d7049de6e7864750bee980c04cdc5f9667/lib/manageiq/api/common/api_error.rb#L8